### PR TITLE
Admin panel campaigns list

### LINKED ISSUE #106: Manage Campaigns (list view, create campaign button)

### PR DESCRIPTION
**Pull request by sivayogasubramanian**
Closes #106

### CODE REVIEW

File: frontend/utils/dates.ts [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function handleDates(body: any) {
+  if (body === null || body === undefined || typeof body !== 'object') return body;
+
+  for (const key of Object.keys(body)) {
+    const value = body[key];
+    if (isIsoDateString(value)) body[key] = moment(value);
+    else if (typeof value === 'object') handleDates(value);
+  }
+}
```
Comment (sivayogasubramanian) [inline frontend/utils/dates.ts:21]: @zognin I know you won't like using REGEX but this makes our types and apis much simpler.

File: frontend/pages/admin/campaigns/index.tsx [hunk 2]
```diff
@@ -1,24 +1,72 @@
...
+  );
   const router = useRouter();
 
+  const upcomingCampaigns = campaigns?.filter((c) => c.start.isAfter(moment.now()));
+  const activeCampaigns = campaigns?.filter((c) => c.start.isBefore(moment.now()) && c.end.isAfter(moment.now()));
+  const completedCampaigns = campaigns?.filter((c) => c.end.isBefore(moment.now()));
+
+  const makeCampaignsTable = (campaigns: Nullable<CampaignAdminListData[]> | undefined) => {
+    const goToManageCampaignAction = {
+      component: <IconButtonWithTooltip icon={<EditIcon />} tooltip="Manage Campaign" />,
```
Comment (zognin) [inline frontend/pages/admin/campaigns/index.tsx:34]: Up to you but maybe we can use this https://mui.com/material-ui/material-icons/?query=manage&selected=ManageAccounts as the button doesn't really go to an edit page

File: frontend/pages/admin/campaigns/index.tsx [hunk 3]
```diff
@@ -1,24 +1,72 @@
...
+          { title: 'ID', key: 'id' },
+          { title: 'Campaign Name', key: 'name' },
+          { title: 'Promised Amount', key: 'promisedAmount' },
+          {
+            title: 'Start Date',
+            key: 'start',
+            transformValue: (start) => start.format(DATE_FORMAT),
+          },
+          {
+            title: 'Start End',
```
Comment (zognin) [inline frontend/pages/admin/campaigns/index.tsx:50]:
```suggestion
            title: 'End Date',
```

**Review by zognin — COMMENTED** [review]
LGTM didn't really check the type changes so hopefully there's no regression.

**Review by zognin — APPROVED** [review]

**Pull request merged**